packages/astro/src/runtime/client/dev-toolbar/apps/audit/a11y.ts

```diff
@@ -496,13 +498,17 @@ export const a11y: AuditRuleWithSelector[] = [
 			if (is_semantic_role_element(role, element.localName, getAttributeObject(element))) {
 				return;
 			}
-			const { requiredProps } = roles.get(role)!;
-			const required_role_props = Object.keys(requiredProps);
-			const missingProps = required_role_props.filter((prop) => !element.hasAttribute(prop));
-			if (missingProps.length > 0) {
-				(element as any).__astro_role = role;
-				(element as any).__astro_missing_attributes = missingProps;
-				return true;
+			
+			const elementRoles = role.split(WHITESPACE_REGEX) as ARIARoleDefinitionKey[];
+			for(const elementRole of elementRoles) {
+				const { requiredProps } = roles.get(elementRole)!;
```

Comment:
This will still fail if the role isn't a real one.

Reply:
> This will still fail if the role isn't a real one.

@Princesseuh Could you please provide the code snippet where you reported an error so that I can verify it again.

I have verified through my blog that when there is a multi role SVG generated by mermaid, errors are no longer reported. The code is as follows:
```html
<svg aria-roledescription="sequence" role="graphics-document document" viewBox="-50 -10 1146 825"
  style="max-width: 1146px;" xmlns="http://www.w3.org/2000/svg" width="100%" id="diagram">
...
</svg>
```
Content written in markdown:
```markdown
```mermaid
sequenceDiagram
...
```.
```

Reply:
Yes, but I think this situation should be modified by the application layer to the real role.

Reply:
Well, I sure hope they do, but I think I'd prefer the entire app to not crash when they don't, ha

Reply:
> Well, I sure hope they do, but I think I'd prefer the entire app to not crash when they don't, ha

@Princesseuh In fact, I am quite confused, because I think that although such code will increase robustness and fault tolerance, in fact, I think that wrong usage should be corrected, not acquiesced.

Reply:
Yes, we can tackle this in a separate PR for sure.

Reply:
Okay, thank you for your guidance.